### ISSUE TITLE
Kubedns is not working correctly on k8s 1.6

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#9120]
**Rancher versions:**
rancher/server: v1.6.2

kubernetes (if applicable): 1.6.6

Kubedns is trying to connect to kubernetes over HTTP and failing:

```6/20/2017 11:41:07 PMI0620 20:41:07.447416       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
6/20/2017 11:41:07 PME0620 20:41:07.542921       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get http://kubernetes.kubernetes:80/api/v1/endpoints?resourceVersion=0: dial tcp 10.42.183.255:80: getsockopt: connection refused
6/20/2017 11:41:07 PME0620 20:41:07.542921       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get http://kubernetes.kubernetes:80/api/v1/services?resourceVersion=0: dial tcp 10.42.183.255:80: getsockopt: connection refused
6/20/2017 11:41:07 PMI0620 20:41:07.947417       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
6/20/2017 11:41:08 PMI0620 20:41:08.447405       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
6/20/2017 11:41:08 PME0620 20:41:08.544386       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get http://kubernetes.kubernetes:80/api/v1/services?resourceVersion=0: dial tcp 10.42.183.255:80: getsockopt: connection refused
6/20/2017 11:41:08 PME0620 20:41:08.544386       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get http://kubernetes.kubernetes:80/api/v1/endpoints?resourceVersion=0: dial tcp 10.42.183.255:80: getsockopt: connection refused
6/20/2017 11:41:08 PMI0620 20:41:08.947407       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
6/20/2017 11:41:09 PMF0620 20:41:09.447393       1 dns.go:168] Timeout waiting for initialization
```

**Comment by deniseschannon** [comment]
Available in v1.6.6-rancher1-4 version of k8s.

**Comment by galal-hussein** [comment]
Rancher server: v1.6.3-rc1
k8s version: rancher/k8s:v1.6.6-rancher1-4

Verified that kube-dns started correctly, and working as expected.
